cmake/ThrustUtilities.cmake

```diff
@@ -11,15 +11,22 @@ function(thrust_wrap_cu_in_cpp cpp_file_var cu_file thrust_target)
   set(${cpp_file_var} "${cpp_file}" PARENT_SCOPE)
 endfunction()
 
-# Enable RDC for a CUDA target. Encapsulates compiler hacks:
-function(thrust_enable_rdc_for_cuda_target target_name)
+# Enable or disable RDC for a CUDA target.
+# Just using the CMake property won't work for our nvcxx builds, we need
+# to manually specify flags.
+# nvcc disables RDC by default, while nvc++ enables it. Thus this function
+# must be called on all CUDA targets to get consistent RDC state across all
+# platforms.
+function(thrust_set_rdc_state target_name enable)
```

Comment:
Any reason we don't roll the `if("CUDA" STREQUAL "${config_device})" in to this function instead of having to guard each call?

Reply:
This would simplify things a bit, but setting RDC state on non-CUDA targets wouldn't make sense so I wanted to avoid using this on non-CUDA targets.